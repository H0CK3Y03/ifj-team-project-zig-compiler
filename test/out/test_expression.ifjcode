.IFJcode24
DEFVAR GF@&discard
CREATEFRAME
CALL main
EXIT int@0

LABEL &strcmp
PUSHFRAME
CREATEFRAME
DEFVAR TF@s1
DEFVAR TF@s2
POPS TF@s2
POPS TF@s1
DEFVAR TF@cnt
DEFVAR TF@res
LT TF@cnt TF@s1 TF@s2
JUMPIFEQ &strcmp_smaller TF@cnt bool@true
GT TF@cnt TF@s1 TF@s2
JUMPIFEQ &strcmp_bigger TF@cnt bool@true
MOVE TF@res int@0
PUSHS TF@res
POPFRAME
RETURN
LABEL &strcmp_smaller
MOVE TF@res int@-1
PUSHS TF@res
POPFRAME
RETURN
LABEL &strcmp_bigger
MOVE TF@res int@1
PUSHS TF@res
POPFRAME
RETURN

LABEL &ord
PUSHFRAME
CREATEFRAME
DEFVAR TF@res
DEFVAR TF@s
DEFVAR TF@i
POPS TF@i
POPS TF@s
DEFVAR TF@len
STRLEN TF@len TF@s
PUSHS int@0
PUSHS TF@len
JUMPIFEQS &ord_err
PUSHS TF@i
PUSHS int@0
LTS
PUSHS bool@true
JUMPIFEQS &ord_err
PUSHS TF@i
PUSHS TF@len
LTS
NOTS
PUSHS bool@true
JUMPIFEQS &ord_err
PUSHS TF@s
PUSHS TF@i
STRI2INTS
POPFRAME
RETURN
LABEL &ord_err
MOVE TF@res int@0
PUSHS TF@res
POPFRAME
RETURN

LABEL &substring
PUSHFRAME
CREATEFRAME
DEFVAR TF@s
DEFVAR TF@i
DEFVAR TF@j
POPS TF@j
POPS TF@i
POPS TF@s
DEFVAR TF@res
MOVE TF@res string@
DEFVAR TF@cnt_err
LT TF@cnt_err TF@i int@0
JUMPIFEQ &substring_err TF@cnt_err bool@true
LT TF@cnt_err TF@j int@0
JUMPIFEQ &substring_err TF@cnt_err bool@true
LT TF@cnt_err TF@j TF@i
JUMPIFEQ &substring_err TF@cnt_err bool@true
DEFVAR TF@len
STRLEN TF@len TF@s
GT TF@cnt_err TF@j TF@len
JUMPIFEQ &substring_err TF@cnt_err bool@true
LT TF@cnt_err TF@i TF@len
NOT TF@cnt_err TF@cnt_err
JUMPIFEQ &substring_err TF@cnt_err bool@true
DEFVAR TF@char
LABEL &substring_loop
JUMPIFEQ &substring_loop_end TF@i TF@j
GETCHAR TF@char TF@s TF@i
CONCAT TF@res TF@res TF@char
ADD TF@i TF@i int@1
JUMP &substring_loop
LABEL &substring_loop_end
PUSHS TF@res
POPFRAME
RETURN
LABEL &substring_err
PUSHS nil@nil
POPFRAME
RETURN

LABEL main
PUSHFRAME
CREATEFRAME
DEFVAR TF@a
DEFVAR TF@b
DEFVAR TF@c
DEFVAR TF@d
DEFVAR TF@result
PUSHS int@5
POPS TF@a
PUSHS float@0x1.4p+1
POPS TF@b
PUSHS int@10
POPS TF@c
PUSHS float@0x1.d333333333333p+2
POPS TF@d
PUSHS TF@a
INT2FLOATS
PUSHS TF@b
ADDS
PUSHS TF@c
PUSHS int@3
SUBS
INT2FLOATS
MULS
PUSHS float@0x1p+1
DIVS
PUSHS TF@d
PUSHS int@3
INT2FLOATS
MULS
ADDS
PUSHS TF@c
PUSHS TF@a
ADDS
PUSHS TF@b
FLOAT2INTS
IDIVS
INT2FLOATS
SUBS
POPS TF@result
PUSHS TF@result
PUSHFRAME
CREATEFRAME
DEFVAR TF@out
POPS TF@out
PUSHS TF@out
WRITE TF@out
POPFRAME
POPFRAME
RETURN
