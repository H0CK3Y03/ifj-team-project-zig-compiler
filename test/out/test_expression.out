0x1.4933333333333p5